cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -437,34 +425,39 @@ private void checkReconfigurability(PodRef podRef, Pod pod, RestartContext resta
                         && "Unschedulable".equals(ps.getReason())
                         && "False".equals(ps.getStatus()));
         if (podStuck
-                && !reasonToRestartPod.contains("Pod has old generation")   // "Pod has old generation" is used with StatefulSets
-                && !reasonToRestartPod.contains("Pod has old revision")) {  // "Pod has old revision" is used with PodSets
+                && !reasonsToRestartPod.contains(RestartReason.POD_HAS_OLD_GENERATION) // "Pod has old generation" is used with StatefulSets
```

Comment:
I wonder if `hasReason` or `shouldRestartFor` might be a better name than `contains`? I think `RestartReasons` should be a richer type than a glorified map. 

It might even read better here to make the method accept multiple reasons so we end up with:
```suggestion
                && !reasonsToRestartPod.shouldRestartFor(RestartReason.POD_HAS_OLD_GENERATION, RestartReason.POD_HAS_OLD_REVISION)
```